src/Plugin/Action/EntityPropertyFetch.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\EntityPropertyFetch.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'Fetch entities by property' action.
+ *
+ * @Action(
+ *   id = "rules_entity_property_fetch",
+ *   label = @Translation("Fetch entities by property"),
+ *   category = @Translation("Entity"),
+ *   context = {
+ *     "type" = @ContextDefinition("string",
+ *       label = @Translation("Entity type"),
+ *       description = @Translation("Specifies the type of the entity that should be fetched."),
+ *     ),
+ *     "property" = @ContextDefinition("string",
+ *       label = @Translation("Property"),
+ *       description = @Translation("The property by which the entity is to be selected.."),
+ *     ),
+ *     "value" = @ContextDefinition("any",
+ *       label = @Translation("Value"),
+ *       description = @Translation("The property value of the entity to be fetched."),
+ *     ),
+ *     "limit" = @ContextDefinition("integer",
+ *       label = @Translation("Limit"),
+ *       description = @Translation("Limit the maximum number of fetched entities."),
+ *       required = FALSE,
+ *     ),
+ *   },
+ *   provides = {
+ *      "entity_fetched" = @ContextDefinition("entity",
+ *        label = @Translation("Fetched entity"),
+ *        multiple = TRUE,
+ *      )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class EntityPropertyFetch extends RulesActionBase implements ContainerFactoryPluginInterface {
```

Comment:
should be called "EntityFetchByField", also as the plugin name and the "property" context should be "field"